src/MountedTraversal.js

```diff
@@ -177,6 +178,12 @@ export function parentsOfInst(inst, root) {
   return pathToNode(inst, root).reverse();
 }
 
+export function instMatchesObjectProps(inst, props) {
+  if (!isDOMComponent(inst)) return false;
+  const node = getNode(inst);
+  return !isEmpty(props) && isMatch(propsOfNode(node), props);
```

Comment:
I too would lean towards throwing an error. `isEmpty` currently exists because an empty object matches all nodes, and that is probably not what a user would want.